tests/gdb-tests/tests/test_command_search.py

```diff
@@ -182,3 +182,37 @@ def test_command_search_rwx(start_binary):
             result_count += 1
 
     assert result_count == 0
+
+
+def test_command_search_asm(start_binary):
+    """
+    Tests searching for asm instructions
+    """
+    start_binary(SEARCH_BINARY)
+
+    gdb.execute("break break_here")
+    gdb.execute("run")
+
+    result_str = gdb.execute('search --asm "add rax, rdx" search_memory', to_string=True)
+    result_count = 0
+    for line in result_str.split("\n"):
+        if line.startswith("search_memory"):
+            result_count += 1
+    assert result_count == 2
```

Comment:
Can we assert th whole expected output instead of checking if line starts with sth etc?
We could use regex for addresses etc.

Reply:
Sorry for delayed response
Will an assertion using this regex be fine?
```
instruction_pattern = re.compile(r"search_memory\.out [A-Za-z0-9]+ add rax, rdx")
```
Can I still make my changes here?